### PR TITLE
FalseDiscoveryRate: removed redundant parameters, simplified tool

### PR DESCRIPTION
**Pull request by timosachsenberg**
performed changes decided in https://github.com/OpenMS/OpenMS/issues/2465

### CODE REVIEW

**Comment by mwalzer** [comment]
:+1:

File: src/topp/FalseDiscoveryRate.cpp [hunk 1]
```diff
@@ -103,14 +103,8 @@ class TOPPFalseDiscoveryRate :
 
   void registerOptionsAndFlags_()
   {
-    registerInputFile_("in", "<file>", "", "Identification input file containing a search against a concatenated sequence database. Either specify '-in' alone or 'in_target' together with 'in_decoy' as input.", false);
+    registerInputFile_("in", "<file>", "", "Identification input file containing a search against a concatenated sequence database. Either specify '-in' alone or 'in_target' together with 'in_decoy' as input.");
```
Comment (cbielow) [inline src/topp/FalseDiscoveryRate.cpp]: `in_*` was removed, right?! Update docu

File: src/topp/FalseDiscoveryRate.cpp [hunk 2]
```diff
@@ -63,11 +63,11 @@ using namespace std;
     </table>
 </CENTER>
 
-    This TOPP tool calculates the false discovery rate (FDR) for results of target-decoy searches. It can handle separate searches of a target database (e.g. forward sequences) and a decoy database (e.g. reversed sequences), or a combined search of a concatenated target-decoy database. The FDR calculation can be performed for proteins and/or for peptides (more exactly, peptide spectrum matches).
+    This TOPP tool calculates the false discovery rate (FDR) for results of target-decoy searches. The FDR calculation can be performed for proteins and/or for peptides (more exactly, peptide spectrum matches).
```
Comment (cbielow) [inline src/topp/FalseDiscoveryRate.cpp]: maybe add a sentence like:

`FalseDiscoveryRate only annotates peptides and proteins with their FDR. A subsequent FDR filtering step needs to be conducted downstream via @ref IDFilter or after exporting the data to a text-based format.`

File: src/topp/FalseDiscoveryRate.cpp [hunk 3]
```diff
@@ -29,7 +29,7 @@
 //
 // --------------------------------------------------------------------------
 // $Maintainer: Chris Bielow $
```
Comment (cbielow) [inline src/topp/FalseDiscoveryRate.cpp]: want to become maintainer? :)

File: src/topp/FalseDiscoveryRate.cpp [hunk 4]
```diff
@@ -167,72 +138,38 @@ class TOPPFalseDiscoveryRate :
...
+      return INCOMPATIBLE_INPUT_DATA;
+    }
 
-      // TODO @all shouldn't ranks be assigned here as well?
-
-      //-------------------------------------------------------------
-      // writing output
-      //-------------------------------------------------------------
-      IdXMLFile().store(out, prot_target, pep_target);
+    for (vector<ProteinIdentification>::iterator it = prot_ids.begin(); it != prot_ids.end(); ++it)
```
Comment (cbielow) [inline src/topp/FalseDiscoveryRate.cpp]: just a note: this bloat code is one of the reasons we should have a dedicated class for `vector<Pep|Prot>` things

**Comment by timosachsenberg** [comment]
can we together scetch out the (hopefully quite minimal) interface of these classes? this might lower the bar for e.g. new members to implement this?

**Comment by timosachsenberg** [comment]
By the way: If you do the changes please just add commits (no rebase)

**Comment by cbielow** [comment]
Should we do that here? The changes will be rather large, since we’d need to replace every occurrence of vector<Pep|ProtId> in the lib, right?

--
Dr. Chris Bielow

Head of Bioinformatics Solution Center (BSC)
Freie Universität Berlin, Institut für Informatik, Takustr. 9, room K21, 14195 Berlin
Phone: +49 (0)30 838-63084, Email: Chris.Bielow(at)fu-berlin.de

**Comment by timosachsenberg** [comment]
no maybe new issue

File: src/topp/FalseDiscoveryRate.cpp (hunk 4 above)
Comment (timosachsenberg) [inline src/topp/FalseDiscoveryRate.cpp]: its not sooo bad if we have auto ;) but I agree that we should have a dedicated, non-template class that aggregates the IDs

**Comment by timosachsenberg** [comment]
I addressed all comments. merging

**Pull request merged**